sector-builder/src/metadata_manager.rs

```diff
@@ -200,9 +200,25 @@ impl<T: KeyValueStore, S: SectorStore> SectorMetadataManager<T, S> {
     }
 
     // Produces a vector containing metadata for all staged sectors that this
-    // SectorBuilder knows about.
-    pub fn get_staged_sectors(&self) -> Result<Vec<StagedSectorMetadata>> {
-        Ok(self.state.staged.sectors.values().cloned().collect())
+    // SectorBuilder knows about. If a sealing status is provided, return only
+    // the staged sector metadata with matching status.
+    pub fn get_staged_sectors(
+        &self,
+        target_status: Option<SealStatus>,
+    ) -> Vec<StagedSectorMetadata> {
+        self.state
+            .staged
+            .sectors
+            .values()
+            .cloned()
```

Comment:
you could reduce the amount of cloning, by doing this after the `.filter`

Reply:
Actually, couldn't this return `Vec<&StagedSectorMetadata>`, as far as I can tell there is no need to clone these.

Reply:
sounds good, thanks!